session/grpc.go

```diff
@@ -79,21 +80,42 @@ func monitorHealth(ctx context.Context, cc *grpc.ClientConn, cancelConn func())
 	defer cancelConn()
 	defer cc.Close()
 
-	ticker := time.NewTicker(1 * time.Second)
+	ticker := time.NewTicker(5 * time.Second)
 	defer ticker.Stop()
 	healthClient := grpc_health_v1.NewHealthClient(cc)
 
+	hasFailedBefore := false
```

Comment:
Nit: simplify name to `failedBefore`?

I think this should reset at some point - maybe after 5 successful health checks?